gtsam/hybrid/HybridValues.h

```diff
@@ -0,0 +1,127 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ *  @file HybridValues.h
+ *  @date Jul 28, 2022
+ *  @author Shangjie Xue
+ */
+
+#pragma once
+
+#include <gtsam/discrete/Assignment.h>
+#include <gtsam/discrete/DiscreteKey.h>
+#include <gtsam/discrete/DiscreteValues.h>
+#include <gtsam/inference/Key.h>
+#include <gtsam/linear/VectorValues.h>
+#include <gtsam/nonlinear/Values.h>
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace gtsam {
+
+/**
+ * HybridValues represents a collection of DiscreteValues and VectorValues.  It
+ * is typically used to store the variables of a HybridGaussianFactorGraph.
+ * Optimizing a HybridGaussianBayesNet returns this class.
+ */
+class GTSAM_EXPORT HybridValues {
+ public:
+  // DiscreteValue stored the discrete components of the HybridValues.
+  DiscreteValues discrete;
+
+  // VectorValue stored the continuous components of the HybridValues.
+  VectorValues continuous;
```

Comment:
Should we name the class `HybridVectorValues`? This class is linear only. (comment only)

Reply:
Maybe I can change VectorValues to Values?

Reply:
I think this is more like a design question: Is this limited to linear hybrid systems? If so we should name it Gaussian.

Reply:
I think we will extend that for nonlinear hybrid later, but currently it's only for linear hybrid system. 
If we rename this to HybridVectorValues, then we will also need HybridValues for nonlinear hybrid system later. Is it necessary to have both HybridVectorValues and HybridValues?

Reply:
This is the spirit for `linear` and `nonlinear` though. because VectorValues and values are different we should probably have the same